pom.xml

```diff
@@ -177,6 +178,28 @@
                     <tag>${project.artifactId}-${project.version}</tag>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>${jdk.version}</source>
+                    <target>${jdk.version}</target>
+                    <encoding>${source.encoding}</encoding>
+                </configuration>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.codehaus.groovy</groupId>
+                        <artifactId>groovy-eclipse-compiler</artifactId>
+                        <version>3.6.0-03</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>org.codehaus.groovy</groupId>
+                        <artifactId>groovy-eclipse-batch</artifactId>
+                        <version>3.0.4-02</version>
+                    </dependency>
```

Comment:
might be a requirement of the parent pom, but looks quite weird to me.

Reply:
@McFoggy So the issue was this...

We need to specify the source and target version of Java to the Maven Compiler Plugin to ensure that we get 1.8. We are now using the maven-integration-plugin to do integration tests. The maven-integration-plugin expects you to write test scripts in Groovy, which we have done.

When the Maven Compiler Plugin is executed, it sees that we have both Java and Groovy code and so chooses to use the groovy-eclipse-compiler so that it can compile both types of code. Unfortunately the Maven Compiler Plugin uses an older version of the groovy-eclipse-compiler which does not like the `<source>1.8</source>` and `<target>1.8</target>` arguments and fails the build. Updating to the latest groovy-eclipse-compiler fixes the issue.

Reply:
@mathieucarbou @adamretter I also use maven-invoker in other projects (see [jgitver-maven-plugin](https://github.com/jgitver/jgitver-maven-plugin) for example) and I do not need to explicitly add groovy support.

Groovy is a dependency of maven-invoker and invoker groovy scripts should be evaluated during IT execution.
In this particular case, this PR I mean, the groovy dependencies are introduced by the parent pom (with old versions of groovy compiler) ; I don't know why they are introduced at that level. Then if removed from this pom the build itself fails. 
The question would be why do we need groovy support for the project ?

FYI: when executiong the build you will see that groovy compiler is trying to find to find groovy scripts in `src/main/groovy` and `src/test/groovy` which is not what is used right now.

Reply:
@McFoggy I am not really sure how to explain it in a different way than I did above. Would it be easier perhaps to have a  call where we share a screen and try to resolve this?

Reply:
@adamretter As I said, I am ok with the PR. I was just saying that the groovy part is useless (because not used) but mandatory due to the missed but forced usage from the parent pom.

See [maven-invoker-plugin](https://maven.apache.org/plugin-developers/plugin-testing.html#maven-invoker-plugin) documentation to realize that invoker tests using groovy do not need groovy compilation (which is normal because groovy in most cases is used as a scripting/runtime feature)

If we want to be clean then we "just" have to open another issue to evaluate the need of keeping the groovy compiler part.

Reply:
@McFoggy Okay... But how would you suggest that I improve this without breaking the build?

Reply:
Let's keep it in the PR first and then tackle it in another issue.

The PR is not about cleaning unused groovy compilers ; let's keep it focused.

Reply:
The parent pom is very old... I had to override some plugins from the parent pom to be able to release the 4.0.rc1 some days ago. But. Ido not see anything related to groovy in the parent pom ? The parent pom is declaring some plugins in the plugin dependency management section but does not enforce its usage. The only plugin called in the release profile is the `groovydoc-maven-plugin`, which could easily be disabled by overriding its phase to `none`.
Where is the problem from the parent pom ?